core/src/test/java/org/elasticsearch/index/query/FuzzyQueryBuilderTests.java

```diff
@@ -181,4 +187,16 @@ public void testParseFailsWithMultipleFields() throws IOException {
         e = expectThrows(ParsingException.class, () -> parseQuery(shortJson));
         assertEquals("[fuzzy] query doesn't support multiple fields, found [message1] and [message2]", e.getMessage());
     }
+
+    public void testDeprecationLoggingHeaders() throws IOException {
+        try (ThreadContext threadContext = new ThreadContext(Settings.EMPTY)) {
```

Comment:
Maybe add the threadContext creation in the base class? and set it / remove it in the before /after test ?